### PR TITLE
do home work

### PR DESCRIPTION
**Pull request by TruongTD1112**

### CODE REVIEW

File: Homework/Truong_Tran/lesson_1/Subsets.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Subsets {
+
+    public static List<List<Integer>> subsets(int[] nums) {
+        int size = nums.length;
+        List<List<Integer>> result = new ArrayList<>();
+        int n = (int) Math.pow(2, size);
```
Comment (quanghoang33) [inline Homework/Truong_Tran/lesson_1/Subsets.java]: 2^n = 1 << n

File: Homework/Truong_Tran/lesson_1/Subsets.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            result.add(subList);
+        }
+        return result;
+    }
+
+    // return binary string with specific length
+    public static String toBinary(int x, int len) {
+        if (len > 0) {
+            return String.format("%" + len + "s",
+                    Integer.toBinaryString(x)).replaceAll(" ", "0");
```
Comment (quanghoang33) [inline Homework/Truong_Tran/lesson_1/Subsets.java:29]: O(logn)

**Review by quanghoang33 — CHANGES_REQUESTED** [review]

File: Homework/Truong_Tran/lesson_1/Subsets.java (hunk 1 above)
Comment (TruongTD1112) [reply Homework/Truong_Tran/lesson_1/Subsets.java]: I understand and I updated

File: Homework/Truong_Tran/lesson_1/Subsets.java (hunk 2 above)
Comment (TruongTD1112) [reply Homework/Truong_Tran/lesson_1/Subsets.java:29]: Yeah, I use built in method in Java to convert decimal number -> binary number

**Pull request merged**